server/resp/src/test/java/org/infinispan/server/resp/RespBlockingCommandsTest.java

```diff
@@ -0,0 +1,195 @@
+      this.cacheMode = CacheMode.LOCAL;
+      this.simpleCache = true;
+      return this;
+   }
+
+   @Override
+   protected String parameters() {
+      return "[simpleCache=" + simpleCache + ", cacheMode=" + cacheMode + "]";
+   }
+
+   @Override
+   protected void amendConfiguration(ConfigurationBuilder configurationBuilder) {
+      if (simpleCache) {
+         configurationBuilder.clustering().cacheMode(CacheMode.LOCAL).simpleCache(true);
+      } else {
+         configurationBuilder.clustering().cacheMode(cacheMode);
+      }
+   }
+
+   private RedisFuture<KeyValue<String, String>> registerBLPOPListener(RedisAsyncCommands<String, String> redis, long timeout, String ... keys) {
+      RedisFuture<KeyValue<String, String>> rf = redis.blpop(timeout, keys);
+      CacheNotifierImpl<?, ?> cni = (CacheNotifierImpl<?, ?>) TestingUtil.extractComponent(cache, CacheNotifier.class);
+      // If there's a listener ok otherwise
+      // if rf is done an error during listener registration has happend
+      // no need to wait anymore. test will fail
+      eventually(() -> cni.getListeners().stream().anyMatch(l -> l instanceof BLPOP.PubSubListener) || rf.isDone());
+      return rf;
+   }
+
+   @Test
+   void testBlpopAsync() throws InterruptedException, ExecutionException, TimeoutException {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      var client = createClient(30000, server.getPort());
+      RedisAsyncCommands<String, String> redis1 = client.connect().async();
+      redis.lpush("keyY", "firstY");
+      redis.rpush("key1", "first", "second", "third");
+      try {
+      var cf = registerBLPOPListener(redis1, 0, "keyZ");
+      // Ensure lpush is after blpop
+         redis.lpush("keyZ", "firstZ");
+         var response = cf.get(10, TimeUnit.SECONDS);
+         assertThat(response.getKey()).isEqualTo("keyZ");
+         assertThat(response.getValue()).isEqualTo("firstZ");
+      } finally {
+         RespTestingUtil.killClient(client);
+      }
+   }
+
+   @Test
+   public void testBlpopFailure() throws InterruptedException, ExecutionException, TimeoutException {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      var tStart = timeService.wallClockTime();
+      assertThat(redis.blpop(10, "nullkey")).isNull();
+      var tEnd = timeService.wallClockTime();
+      // The following assertion fails when run with mvn test. Works in IDE
+      // assertThat(tEnd-tStart).isGreaterThanOrEqualTo(5_000L);
+   }
+
+   @Test
+   public void testBlpop() throws InterruptedException, ExecutionException, TimeoutException, java.util.concurrent.TimeoutException {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      var client = createClient(30000, server.getPort());
+      RedisAsyncCommands<String, String> redisBlock = client.connect().async();
+      try {
+         var cf = registerBLPOPListener(redisBlock, 0, "keyZ");
+         Thread.sleep(2000);
```

Comment:
Since the method above ensures the listener is installed, we can remove the sleeps.